From 94a8758ca124e7d04c208eed7713a45ad5d74332 Mon Sep 17 00:00:00 2001
From: Haikel Guemar <hguemar@fedoraproject.org>
Date: Wed, 4 Oct 2017 19:22:58 +0200
Subject: [PATCH 2/2] Drop OpenSSL.rand support

PyOpenSSL deprecated OpenSSL.rand in 17.2.0 and removed it in 17.3.0.
To comply, update min version in tests to 17.3.0 and removes it.
https://pyopenssl.org/en/stable/changelog.html

(cherry picked from commit 5b8f5f595624bdfb5f707707959977bb56864e0d)
---
 eventlet/green/OpenSSL/__init__.py | 1 -
 eventlet/green/OpenSSL/rand.py     | 1 -
 tests/openssl_test.py              | 1 -
 3 files changed, 3 deletions(-)
 delete mode 100644 eventlet/green/OpenSSL/rand.py

diff --git a/eventlet/green/OpenSSL/__init__.py b/eventlet/green/OpenSSL/__init__.py
index 26b60d9..d861476 100644
--- a/eventlet/green/OpenSSL/__init__.py
+++ b/eventlet/green/OpenSSL/__init__.py
@@ -1,4 +1,3 @@
-from . import rand
 from . import crypto
 from . import SSL
 from . import tsafe
diff --git a/eventlet/green/OpenSSL/rand.py b/eventlet/green/OpenSSL/rand.py
deleted file mode 100644
index c670f73..0000000
--- a/eventlet/green/OpenSSL/rand.py
+++ /dev/null
@@ -1 +0,0 @@
-from OpenSSL.rand import *
diff --git a/tests/openssl_test.py b/tests/openssl_test.py
index a4498dc..a127408 100644
--- a/tests/openssl_test.py
+++ b/tests/openssl_test.py
@@ -12,6 +12,5 @@ def test_import():
 
     import eventlet.green.OpenSSL.SSL
     import eventlet.green.OpenSSL.crypto
-    import eventlet.green.OpenSSL.rand
     import eventlet.green.OpenSSL.tsafe
     import eventlet.green.OpenSSL.version
-- 
2.13.5

